### PR TITLE
[wasi] Select mono components

### PR DESCRIPTION
**Pull request by maraf**
- Never link with `diagnostics_tracing`
- Use stubs for `debugger` and `hot_reload` for publish, use real components for build (aligns with browser debugger support)
  - Overridable by explicitly setting `WasmDebugLevel=1` or `0`

Contributes to https://github.com/dotnet/runtime/issues/96862

### CODE REVIEW

**Comment by maraf** [comment]
@lewing Is this what you were refering to yesterday https://github.com/dotnet/runtime/blob/main/src/mono/msbuild/common/RuntimeComponentManifest.targets ?

**Comment by maraf** [comment]
In common targets we have this selection. The condition doesn't follow other logic for enabling debugging, but otherwise it should trigger the logic even for wasi...

https://github.com/dotnet/runtime/blob/main/src/mono/wasm/build/WasmApp.Common.targets#L255-L257

**Comment by maraf** [comment]
> src/mono/wasi/build/WasiApp.targets(264,11): error MSB4057: (NETCORE_ENGINEERING_TELEMETRY=Build) The target "_MonoSelectRuntimeComponents" does not exist in the project.

Seems like MonoTargets are not packed for lib tests..

File: src/mono/wasi/build/WasiApp.targets [hunk 1]
```diff
@@ -23,6 +23,16 @@
       _WasmGenerateNodeScripts;
     </WasmGenerateAppBundleDependsOn>
 
+    <WasiWriteRspForLinkingDependsOn>
```
Comment (lewing) [inline src/mono/wasi/build/WasiApp.targets:26]: where is this used?

Comment (lewing) [reply src/mono/wasi/build/WasiApp.targets:26]: I see now

**Review by lewing — APPROVED** [review]

**Pull request merged**